akvo/rsr/spa/app/modules/editor/project-init-handler.js

```diff
@@ -64,26 +66,30 @@ const ProjectInitHandler = connect(({editorRdr}) => ({ editorRdr }), actions)(Re
     }
     else resolve()
   })
-  let nextSectionIndex = 0
-  const fetchNextSection = () => {
-    fetchSection(sectionIndexPipeline[nextSectionIndex])
-    .then(() => {
-      props.setSectionFetched(sectionIndexPipeline[nextSectionIndex])
-      if (nextSectionIndex < 10) {
-        nextSectionIndex += 1
-        fetchNextSection()
-      }
-    })
-  }
+
   useEffect(() => {
-    if (prevParams && prevParams.id !== params.id && params.id !== 'new'){
-      fetchSection(3)
-      fetchSection(5)
+    // eslint-disable-next-line no-restricted-globals
+    if (!isNaN(params?.id) && (nextSectionIndex < 11)) {
+      if (!props?.editorRdr[`section${nextSectionIndex + 1}`]?.isFetched) {
+        fetchSection(sectionIndexPipeline[nextSectionIndex])
+          .then(() => {
+            if (nextSectionIndex === 4) {
```

Comment:
A constant here too instead of `4` would be better e.g `SECTION_RESULTS_INDICATORS`.

Reply:
Got it! will do.